// e-commerce-api/ecommerce.js
exports.products= [

  {
    id:"1",
    product: "cacio",
    amount: 5

  },
  {
    id:"2",
    product: "provola",
    amount: 6
  },
  {
    id:"3",
    product: "mozzarella",
    amount: 3
  },

  {
    id:"4",
    product: "svizzero",
    amount: 2
  },

  {
    id:"5",
    product: "parmiggiano",
    amount: 6
  },
  {
    id:"6",
    product: "prosciutto",
    amount: 6
  },
  {
    id:"7",
    product: "salame",
    amount: 6
  }
];
exports.buyed=[];
exports.add= function(products,product,amount) {

    var temp=parseInt(products[products.length-1].id)+1;
    var id=""+temp;
    products.push({id:id, product:product, amount:amount});
    return "Product Add";
}
exports.up= function(products,id,product,amount) {
  for (var i = 0; i < products.length; i++) {
    if (id===products[i].id) {
      products[i].product=product;
      products[i].amount=amount;
      return "Product put";
    }

  }


}
exports.del= function(products,id) {
  for (var i = 0; i < products.length; i++) {
    if(id===products[i].id)
    {
      products.splice(i,1);
      return "Product delete";
    }
  }

}

exports.buy= function(buyed,products,idp,user) {
  var cond=false;
  for (var i=0; i<products.length; i++) {

      if(idp===products[i].id)
      {
        cond=true;
        if(products[i].amount>0) {
        products[i].amount-=1;
        buyed.push(
                    {
                      idproduct: products[i].id,
                      timestamp: new Date().getTime(),
                      user: user
                    }
                  );
          return "Buy succesfull, remain "+ products[i].amount+" amount";
        }else return "Products finished";
      }
    }
  if(cond===false){
      return "Invalid ID";
    }
}

/*console.log({"ADD:":add("Mortadella",5)});
console.log({"UP:":up("7","Porchetta",6)});
console.log({"DEL":del("5")});
console.log({"BUY":buy("6")});*/
